dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavTool.java

```diff
@@ -0,0 +1,301 @@
+                        ident = APILocator.getIdentifierAPI()
+                            .find(itemFile.getPermissionId());
+                        NavResult nav = new NavResult(folder.getInode(), host.getIdentifier(), languageId);
+
+                        nav.setTitle(itemFile.getFriendlyName());
+                        nav.setHref(ident.getURI());
+                        nav.setOrder(itemFile.getMenuOrder());
+                        nav.setType("file");
+                        nav.setPermissionId(itemFile.getPermissionId());
+                        nav.setShowOnMenu(itemFile.isShowOnMenu());
+                        nav = new NavResultHydrated(nav, this.context);
+                        children.add(nav);
+                    }
+                }
+            }
+
+            CacheLocator.getNavToolCache()
+                .putNav(host.getIdentifier(), folder.getInode(), result, languageId);
+
+            return new NavResultHydrated(result, this.context);
+        }
+    }
+
+    /**
+     * Pass the level of the nav you wish to retrieve, based on the current path, level 0 being the
+     * root
+     * 
+     * @param level
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    public NavResult getNav(int level) throws DotDataException, DotSecurityException {
+        if (level < 1)
+            return getNav("/");
+
+
+        String reqPath = getNav().getHref();
+
+        String[] levels = reqPath.split("/");
+
+
+        if (level + 1 > levels.length)
+            return null;
+
+        StringBuffer sw = new StringBuffer();
+
+        for (int i = 1; i <= level; i++) {
+            sw.append("/");
+            sw.append(levels[i]);
+        }
+        String path = sw.toString();
+
+        return getNav(path);
+    }
+
+    public NavResult getNav() throws DotDataException, DotSecurityException {
+        return getNav((String) request.getAttribute("javax.servlet.forward.request_uri"));
+    }
+
+    public NavResult getNav(String path) throws DotDataException, DotSecurityException {
+
+        Host host = getHostFromPath(path);
+
+        if (host == null)
+            host = currenthost;
+
+        return getNav(host, path);
+    }
+
+    public NavResult getNav(String path, long languageId) throws DotDataException, DotSecurityException {
+
+        Host host = getHostFromPath(path);
+
+        if (host == null)
+            host = currenthost;
+
+        return getNav(host, path, languageId, systemUser);
+    }
+
+    private Host getHostFromPath(String path) throws DotDataException, DotSecurityException {
+        if (path.startsWith("//")) {
+            List<RegExMatch> find = RegEX.find(path, "^//(\\w+)/(.+)");
+            if (find.size() == 1) {
+                String hostname = find.get(0)
+                    .getGroups()
+                    .get(0)
+                    .getMatch();
+                path = "/" + find.get(0)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "path". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this hard-coded path-delimiter. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1075)